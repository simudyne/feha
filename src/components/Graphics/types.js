export const FETCH_COMMITS_FAILURE = 'FETCH_COMMITS_FAILURE';
export const FETCH_COMMITS_REQUEST = 'FETCH_COMMITS_REQUEST';
export const FETCH_COMMITS_SUCCESS = 'FETCH_COMMITS_SUCCESS';
