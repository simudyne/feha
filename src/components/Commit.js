import * as React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { Segment, Grid, Header, Button, Icon } from 'semantic-ui-react'
import { Committer } from './Committer'

const Commit = ({ data }) => (
  <Segment>
    <Grid centered columns={2} verticalAlign="middle">
      <Grid.Column mobile={16} computer={6}>
        <Header size="tiny">{data.commit.message}</Header>
        <Committer {...data} />
      </Grid.Column>
      <Grid.Column mobile={16} computer={10} textAlign="right">
        {data.commit.verification.verified &&
          <Button basic color='green'>
            Verified
          </Button>}
        <Button
          content='Comments'
          icon='comment'
          label={{ as: 'span', basic: true, content: data.commit.comment_count }}
          labelPosition='right'
        />
        <Button primary>
          <Icon name='copy' />
          {data.sha.substring(0, 7)}
        </Button>
        <Button style={{minWidth: '190px'}}>
          <Icon name='calendar' />
          {moment(data.commit.committer.date).format('LL')}
        </Button>
      </Grid.Column>
    </Grid>
  </Segment>
)

Commit.propTypes = {
  data: PropTypes.shape({
    sha: PropTypes.string.isRequired,
    commit: PropTypes.shape({
      author: PropTypes.shape({
        name: PropTypes.string.isRequired,
        email: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired
      }).isRequired,
      committer: PropTypes.shape({
        name: PropTypes.string.isRequired,
        email: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired
      }).isRequired,
      message: PropTypes.string.isRequired,
      comment_count: PropTypes.number.isRequired,
      verification: PropTypes.shape({
        verified: PropTypes.bool.isRequired
      }).isRequired
    }).isRequired.isRequired,
    html_url: PropTypes.string.isRequired,
    author: PropTypes.shape({
      login: PropTypes.string.isRequired,
      avatar_url: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired
    })
  }).isRequired
}

export default Commit
