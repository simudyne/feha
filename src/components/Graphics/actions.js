import * as types from './types';

export const fetchCommitsRequest = () => ({
  type: types.FETCH_COMMITS_REQUEST
});

export const fetchCommitsSuccess = ticks => ({
  type: types.FETCH_COMMITS_SUCCESS,
  commits
});

export const fetchCommitsFailure = error => ({
  type: types.FETCH_COMMITS_FAILURE,
  error
});
