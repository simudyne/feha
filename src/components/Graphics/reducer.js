import * as types from './types';

const initialState = {
  data: [],
  error: null,
  fetching: false
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case types.FETCH_COMMITS_REQUEST:
      return { ...state, fetching: true, error: null };

    case types.FETCH_COMMITS_SUCCESS:
      return { ...state, fetching: false, data: action.commits };

    case types.FETCH_COMMITS_FAILURE:
      return { ...state, fetching: false, error: action.error };

    default:
      return state;
  }
};

export default reducer;
