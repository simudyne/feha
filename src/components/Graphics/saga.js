import { all, call, put, takeLatest } from 'redux-saga/effects';
import * as types from './types';

export function* fetchCommits() {
  const commits = yield call(requestCommits);
  yield put({ type: types.FETCH_COMMITS_SUCCESS, commits });
}

export function requestCommits() {
  return [{ sha: '👹' }, { sha: '👐' }, { sha: '🦵🦵 ' }];
}

export const commitsSaga = [
  takeLatest(types.FETCH_COMMITS_REQUEST, fetchCommits)
];

export function* rootSaga() {
  yield all([...commitsSaga]);
}
