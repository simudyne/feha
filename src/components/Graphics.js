import * as React from 'react'
import { Grid, Container, Header, Form, Message } from 'semantic-ui-react'

import Commit from './Commit'

const style = {
  height: '100vh',
  fontSize: '24px',
  padding: '1rem'
}

export class Graphics extends React.Component {
  constructor(props) {
    super(props)
    this.state = { repository: '' }
    this.onChange = this.onChange.bind(this)
    this.onSubmit = this.onSubmit.bind(this)
  }

  render() {

    const searchForm = (
      <Form onSubmit={this.onSubmit}>
        <Form.Input
          label='Search a repository...'
          type='text'
          name="repository"
          value={this.state.repository}
          onChange={this.onChange}
          loading={this.props.commits.fetching}
          icon='search'
          iconPosition='left'
          placeholder='facebook/react'
          required
        />
      </Form>
    )

    const header = this.props.commits.data && this.props.commits.data.length > 0 &&
      <Header size="medium" textAlign="center">
        Commits on <a href={`https://github.com/${this.props.commits.repository}`}
                      target="_blank">{this.props.commits.repository}</a>
      </Header>

    const data = this.props.commits.error ?
      <Message negative>
        <Message.Header>{this.props.commits.error.message}</Message.Header>
      </Message> :
      this.props.commits.data.map(commit => (
        <Commit key={commit.sha} data={commit} />
      ))

    return (
      <Container>
        <Grid centered columns={1} style={style} verticalAlign="top">
          <Grid.Column>
            {searchForm}
            {header}
            {data}
          </Grid.Column>
        </Grid>
      </Container>
    )
  }

  onChange(event, {name, value}) {
    this.setState({[name]: value})
  }

  onSubmit() {
    console.log(this.state.repository)
    this.props.getCommits(this.state.repository)
  }
}
