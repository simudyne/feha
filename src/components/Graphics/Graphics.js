import * as React from 'react';
import { connect } from 'react-redux';
import { Segment, Grid } from 'semantic-ui-react';
import { fetchCommitsRequest } from './actions';
import styled from 'styled-components';

const Container = styled.div`
  height: '100vh',
  fontSize: '24px'
  border: 1px solid green;
`;

export class Graphics extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getCommits();
  }

  render() {
    return (
      <Container>
        <Grid centered columns={2} verticalAlign="middle">
          <Grid.Column textAlign="center">
            {this.props.commits.data.map((d, i) => (
              <div key={i}>{d.sha}</div>
            ))}
          </Grid.Column>
        </Grid>
      </Container>
    );
  }
}

const mapStateToProps = state => ({
  commits: state.commits
});

const mapDispatchToProps = dispatch => ({
  getCommits: () => dispatch(fetchCommitsRequest())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Graphics);
