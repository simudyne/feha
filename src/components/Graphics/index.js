import Graphics from './Graphics';
export * from './actions';
export * from './reducer';

export default Graphics;
