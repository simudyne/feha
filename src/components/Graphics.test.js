import * as React from 'react'
import { shallow, mount } from 'enzyme'
import { Form, Header, Message } from 'semantic-ui-react'
import { Graphics } from './Graphics'
import { Commit } from './Commit'
import renderer from 'react-test-renderer'
import sampleData from '../data/commits'

const props = {
  commits: [],
  fetching: false,
  repository: null,
  error: null,
  getCommits: jest.fn()
}

describe('Graphics component', () => {

  it('Should render correctly on initial state', () => {
    const tree = renderer.create(<Graphics {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly while commits are being requested', () => {
    const fetching = {...props, fetching: true}
    const tree = renderer.create(<Graphics {...fetching} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly when commits request is successful', () => {
    const success = {...props, commits: sampleData, repository: 'repo'}
    const tree = renderer.create(<Graphics {...success} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly when commits request is failed', () => {
    const failure = {...props, error: 'error'}
    const tree = renderer.create(<Graphics {...failure} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('getCommits has been called on form submit event', () => {
    const wrapper = shallow(<Graphics {...props} />)
    wrapper.find(Form).simulate('submit')
    expect(props.getCommits).toHaveBeenCalled()
    expect(props.getCommits.mock.calls.length).toBe(1)
    wrapper.find(Form).simulate('submit')
    expect(props.getCommits.mock.calls.length).toBe(2)
  })

  it('header is toggled depending on repository prop value', () => {
    const header = {...props, repository: 'repo'}
    const wrapper = mount(<Graphics {...header} />)
    expect(wrapper.find(Header).exists()).toBe(true)
    expect(wrapper.find(Header).text()).toBe('Commits on repo')
    wrapper.setProps({repository: null})
    expect(wrapper.find(Header).exists()).toBe(false)
  })

  it('error message is toggled depending on error prop value', () => {
    const error = {...props, error: 'error'}
    const wrapper = mount(<Graphics {...error} />)
    expect(wrapper.find(Message).exists()).toBe(true)
    expect(wrapper.find(Message).text()).toBe('error')
    wrapper.setProps({error: null})
    expect(wrapper.find(Message).exists()).toBe(false)
  })

  it('commits graphic is toggled depending on commits prop value', () => {
    const graphic = {...props, commits: sampleData}
    const wrapper = mount(<Graphics {...graphic} />)
    expect(wrapper.find(Commit).exists()).toBe(true)
    expect(wrapper.find(Commit).length).toBe(13)
    wrapper.setProps({commits: null})
    expect(wrapper.find(Message).exists()).toBe(false)
  })
})