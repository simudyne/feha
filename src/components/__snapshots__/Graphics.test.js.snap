// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphics component Should render correctly on initial state 1`] = `
<div
  className="ui container"
>
  <div
    className="ui centered top aligned one column grid"
    style={
      Object {
        "fontSize": "24px",
        "height": "100vh",
        "padding": "1rem",
      }
    }
  >
    <div
      className="column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="required field"
        >
          <label>
            Search a repository...
          </label>
          <div
            className="ui left icon input"
          >
            <input
              name="repository"
              onChange={[Function]}
              placeholder="facebook/react"
              required={true}
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Graphics component Should render correctly when commits request is failed 1`] = `
<div
  className="ui container"
>
  <div
    className="ui centered top aligned one column grid"
    style={
      Object {
        "fontSize": "24px",
        "height": "100vh",
        "padding": "1rem",
      }
    }
  >
    <div
      className="column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="required field"
        >
          <label>
            Search a repository...
          </label>
          <div
            className="ui left icon input"
          >
            <input
              name="repository"
              onChange={[Function]}
              placeholder="facebook/react"
              required={true}
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
            />
          </div>
        </div>
      </form>
      <div
        className="ui negative message"
      >
        <div
          className="header"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          error
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Graphics component Should render correctly when commits request is successful 1`] = `
<div
  className="ui container"
>
  <div
    className="ui centered top aligned one column grid"
    style={
      Object {
        "fontSize": "24px",
        "height": "100vh",
        "padding": "1rem",
      }
    }
  >
    <div
      className="column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="required field"
        >
          <label>
            Search a repository...
          </label>
          <div
            className="ui left icon input"
          >
            <input
              name="repository"
              onChange={[Function]}
              placeholder="facebook/react"
              required={true}
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
            />
          </div>
        </div>
      </form>
      <div
        className="ui medium center aligned header"
      >
        Commits on 
        <a
          href="https://github.com/repo"
          target="_blank"
        >
          repo
        </a>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Remove styled-components dependency as it wasn't being used extensively
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              1f5979c
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 11, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Tests for commits reducer
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              8a76880
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Configure testing suite (jest + enzyme)
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              a0f721a
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Remove log
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              8b9ae34
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Search commits by repository feature
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              9e75cc3
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Complete Commit component data and layout
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                2
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              931263d
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Changes in Commit component layout
            </div>
            <div
              className="commit"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://cdn1.iconfinder.com/data/icons/user-pictures/100/unknown-512.png"
              />
              <a
                href="mailto:juan.lloretmontalvo@zooplus.com"
              >
                Juan Lloret
              </a>
               authored and 
              <a
                href="mailto:juan.lloretmontalvo@zooplus.com"
              >
                Juan Lloret
              </a>
               committed
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              0757c7f
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Include styled-components in the project
            </div>
            <div
              className="commit"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://cdn1.iconfinder.com/data/icons/user-pictures/100/unknown-512.png"
              />
              <a
                href="mailto:juan.lloretmontalvo@zooplus.com"
              >
                Juan Lloret
              </a>
               authored and 
              <a
                href="mailto:juan.lloretmontalvo@zooplus.com"
              >
                Juan Lloret
              </a>
               committed
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              79c1dff
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Include Loader component while fetching
            </div>
            <div
              className="commit"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://cdn1.iconfinder.com/data/icons/user-pictures/100/unknown-512.png"
              />
              <a
                href="mailto:juan.lloretmontalvo@zooplus.com"
              >
                Juan Lloret
              </a>
               authored and 
              <a
                href="mailto:juan.lloretmontalvo@zooplus.com"
              >
                Juan Lloret
              </a>
               committed
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              43ee03c
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Perform fetch request and first draft for Commit component
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              99cca1b
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Add .idea folder to gitignore
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars2.githubusercontent.com/u/7203420?v=4"
              />
              <a
                href="https://github.com/juallom"
                target="_blank"
              >
                juallom
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              9f46226
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              January 10, 2019
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              🚧 scaffolding goes up 🏗️
            </div>
            <div
              className="commit"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://cdn1.iconfinder.com/data/icons/user-pictures/100/unknown-512.png"
              />
              <a
                href="mailto:kristian@kristians-macbook.local"
              >
                Kristian
              </a>
               authored and 
              <a
                href="mailto:kristian@kristiasmacbook.simudyne"
              >
                Kristian
              </a>
               committed
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              58fcc69
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              November 28, 2018
            </button>
          </div>
        </div>
      </div>
      <div
        className="ui segment"
      >
        <div
          className="ui centered middle aligned two column grid"
        >
          <div
            className="six wide computer sixteen wide mobile column"
          >
            <div
              className="ui tiny header"
            >
              Initial commit
            </div>
            <div
              className="author"
              style={
                Object {
                  "color": "#586069",
                  "fontSize": ".8rem",
                }
              }
            >
              <img
                className="ui avatar image"
                src="https://avatars0.githubusercontent.com/u/5860402?v=4"
              />
              <a
                href="https://github.com/kristianKan"
                target="_blank"
              >
                kristianKan
              </a>
            </div>
          </div>
          <div
            className="right aligned ten wide computer sixteen wide mobile column"
          >
            <button
              className="ui green basic button"
              onClick={[Function]}
              role="button"
            >
              Verified
            </button>
            <div
              className="ui right labeled button"
              onClick={[Function]}
            >
              <button
                className="ui icon button"
                tabIndex={0}
              >
                <i
                  aria-hidden="true"
                  className="comment icon"
                />
                 
                Comments
              </button>
              <span
                className="ui left pointing basic label"
                onClick={[Function]}
              >
                0
              </span>
            </div>
            <button
              className="ui primary button"
              onClick={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="copy icon"
              />
              40724f9
            </button>
            <button
              className="ui button"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "minWidth": "190px",
                }
              }
            >
              <i
                aria-hidden="true"
                className="calendar icon"
              />
              November 27, 2018
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Graphics component Should render correctly while commits are being requested 1`] = `
<div
  className="ui container"
>
  <div
    className="ui centered top aligned one column grid"
    style={
      Object {
        "fontSize": "24px",
        "height": "100vh",
        "padding": "1rem",
      }
    }
  >
    <div
      className="column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="required field"
        >
          <label>
            Search a repository...
          </label>
          <div
            className="ui loading left icon input"
          >
            <input
              name="repository"
              onChange={[Function]}
              placeholder="facebook/react"
              required={true}
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
